### PR TITLE
Passport Local - login(), logout(), register() GraphQL mutations

### PR DESCRIPTION
**Pull request by AliceRossa**
Integration of Passport-local for React Starker Kit.

**Help wanted**
- Token not generated
- Unprotected Profile -> if (user.Authenticated) ?
- Error Processing (Invalid credentials, Password 8 characters long...)
- No page redirection
- Logout

### CODE REVIEW

File: src/data/mutations/user.login.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+
+    const usernameOrEmailLC = usernameOrEmail.toLowerCase();
+    const findUser = await User.findOne({
+      attributes: ['id', 'username', 'email'],
+      where: {
+        $or: [{ username: usernameOrEmailLC }, { email: usernameOrEmailLC }],
+      },
+    });
+
+    if (user && user.comparePassword(password)) {
```
Comment (chunsli) [inline src/data/mutations/user.login.js]: wouldn't user be null? Do you mean findUser here?

File: src/data/models/User.js [hunk 2]
```diff
@@ -28,12 +35,23 @@ const User = Model.define('User', {
...
 
+  password: {
+    type: DataType.STRING,
+    allowNull: false,
+  },
+
 }, {
 
+  instanceMethods: {
+    comparePassword: (password) => bcrypt.compareSync(password, this.password),
```
Comment (chunsli) [inline src/data/models/User.js]: the scope here will be wrong for `this.password` with an arrow function. I think you need this:
`comparePassword: function(password){return bcrypt.compareSync(password, this.password)}`

File: src/data/mutations/user.logout.js [hunk 3]
```diff
@@ -0,0 +1,10 @@
+import UserType from '../types/UserType';
+
+const userLogout = {
+  type: UserType,
+  resolve({ res }) {
```
Comment (stfnh) [inline src/data/mutations/user.logout.js]: i think you forgot to pass res in src/server.js

File: src/server.js [hunk 4]
```diff
@@ -54,19 +54,32 @@ app.use(expressJwt({
...
+      console.log(e); // eslint-disable-line no-console
+    }
+  }
+  next();
+});
 app.use(passport.initialize());
 
+const handleAuth = (req, res) => {
+  const token = jwt.sign(req.user, auth.jwt.secret, { expiresIn: auth.jwt.expires });
+  res.cookie('id_token', token, { maxAge: 1000 * auth.jwt.expires, httpOnly: true });
```
Comment (stfnh) [inline src/server.js]: didn't you mean
`res.cookie('id-token', ...`
hyphen instead of underscore

**Comment by TylerL-uxai** [comment]
@AliceRossa Thank you! Does this register/log on work fully?

**Pull request closed without merging**